### PR TITLE
feat(poll): expose custom poll interval

### LINKED ISSUE #13775: [Feature] Allow to define an interval / wait period between expect.poll assertions
In one of the latest versions of Playwright the `expect.poll`-function was introduced to allow to poll for a specific assertion. This is really a great addition to Playwright. 

Only I would like to request an enhancement to this functionality to delay a breathing period between assertion checks. For example, when you are querying for a status change via an API endpoint then it can be beneficial to wait a short period between assertion checks to not hit the API server too often or it might not beneficial to continuously execute this assertion.

In my case I would like to check if a revocation of permissions has completed, and it doesn't make sense to check the assertion of a period of 60 seconds continuously. A bit of breathing or waiting time of like 5 seconds would help to ease everything

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes https://github.com/microsoft/playwright/issues/13775

### CODE REVIEW

File: packages/playwright-test/src/expect.ts [hunk 1]
```diff
@@ -248,10 +250,11 @@ class ExpectMetaInfoProxyHandler {
   }
 }
 
-async function pollMatcher(matcherName: any, isNot: boolean, timeout: number, generator: () => any, ...args: any[]) {
+async function pollMatcher(matcherName: any, isNot: boolean, pollIntervals: number[] | undefined, timeout: number, generator: () => any, ...args: any[]) {
   let matcherError;
   const startTime = monotonicTime();
-  const pollIntervals = [100, 250, 500];
+  pollIntervals = pollIntervals || [100, 250, 500, 1000];
```
Comment (aslushnikov) [inline packages/playwright-test/src/expect.ts:256]:
```suggestion
  pollIntervals = pollIntervals ?? [100, 250, 500, 1000];
```

File: packages/playwright-test/src/expect.ts [hunk 2]
```diff
@@ -248,10 +250,11 @@ class ExpectMetaInfoProxyHandler {
...
+  const lastPollInterval = pollIntervals[pollIntervals.length - 1] || 1000;
```
Comment (aslushnikov) [inline packages/playwright-test/src/expect.ts:257]: I assume if someone passes [0] as poll intervals we want to respect it
```suggestion
  const lastPollInterval = pollIntervals[pollIntervals.length - 1] ?? 1000;
```

**Review by aslushnikov — APPROVED** [review]

File: packages/playwright-test/src/expect.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/playwright-test/src/expect.ts:256]: This should be the same thing.

File: packages/playwright-test/src/expect.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/playwright-test/src/expect.ts:257]: I'm not sure we ever want to see 0 here? That'd be a denial of service more or less...

**Pull request merged**